### ISSUE TITLE
Add support for for xml schema as a model definition

### ISSUE THREAD

**Issue by jahlborn** [OAI/OpenAPI-Specification#54]
I'm looking for a way to provide metadata for a REST webservice, and swagger is currently my leading choice. The biggest missing feature (for me) is the ability to use xml schema to describe a model. i understand this is potentially a large feature. even if it may not be added anytime soon, it would be great if there was at least a rough "proposal" for a good way that the current spec could be modified to support xml schema.

**Comment by rhyde** [comment]
This is very unfortunate that there appears to be no interest in being compatible with existing standards.

**Comment by darrelmiller** [comment]
@rhyde It is unfortunate that you have got that impression.  Fortunately it is an inaccurate impression.  We care very deeply about being compatible with existing standards.  We also care about ensuring that tooling that implements the OpenAPI spec provides a consistent experience for users.  Supporting schemas other than JSON Schema is something that many of us in the OpenAPI community want to be able to do.  However, we need to do it in a way that meets all of our other goals and objectives.  We were not able to do that in the V3 timeline.  We definitely will be re-addressing this issue as we move forward and we look forward to your input and feedback on this important feature.

**Comment by sendtonirav** [comment]
Is there a tool that lets you automatically and reliably convert the XML schema (XSD) to the swagger supported JSON schema? Atleast that way I am managing one schema and then just auto converting it to the supported format?

**Comment by MikeRalphson** [comment]
From this [stackoverflow answer](https://stackoverflow.com/questions/3922026/generate-json-schema-from-xml-schema-xsd) there are a few options mentioned.

A tool I wrote called [jgeXml](https://github.com/mikeralphson/jgexml) has a module for converting XSD to JSON schema (and particularly OpenAPI-flavour JSON schema). It isn't an exact 1:1 conversion, though it has been used to convert a complex 1,700+ line [XML schema](https://github.com/Mermade/bbcparse/blob/master/nitroApi/nitro-schema.xsd) to build valid OpenAPI 2.0 [schema objects](https://github.com/Mermade/bbcparse/blob/master/nitroApi/nitro-schema.json), Feedback is welcome.

**Comment by benp444** [comment]
OK, so given that there is no support for XML in Swagger, what is the commonly accepted work around where:

a) your data models are defined outside of the Swagger files and
b) you data models are defined in XSD.

My clumsy efforts have led me to the following Swagger file where I am using description to indicate to developers what they must expect in terms of payload definitions:

```
 post:
      summary: Creates a new student
      parameters:
        - name: StudentPersonUBO
          in: body
          description: 'http://uwe.ac.uk/schema/ubo/student/StudentPersonUBO.xsd'
          schema:
            type: string
```
There must be some commonly accepted practice for using Swagger whilst at the same time stating _"the payload should be based on this XML schema"_

Can anyone provide some insight?

**Comment by ioggstream** [comment]
Work-in-progress here https://github.com/OAI/OpenAPI-Specification/pull/1736

**Comment by slogic** [comment]
ok, that thread is closed until oas 3.1.0 is ready. i see it is published 2 years ago. where is continuation of this thread?
